src/Silverback.Integration.MQTT/Messaging/Broker/Mqtt/MqttClientWrapper.cs

```diff
@@ -92,6 +94,13 @@ public Task DisconnectAsync(object sender)
 
             lock (_connectionLock)
             {
+                if (ClientConfig.OnDisconnecting != null &&
```

Comment:
I think that this should be moved down, right before the actual call to `MqttClient.DisconnectAsync`.